main.cpp

```diff
@@ -13,75 +13,495 @@ What you create in this project will be used as the basis of Project 5 in the co
+ */
+struct sipProvider
+{
+    //supported codecs
+    std::string codecName = "G729";
+    //number of simultaneous calls
+    int calls=2;
+    //FAX support protocol
+    bool t38 = false;
+    //price
+    double price=0;
+    //SLA conditions
+    int slaType = 1;
+    
+    //do calls
+    bool makeCall(int src, int dst);
+    //charge customer
+    float chargeCustomer(double time, int customerId);
+    //lease traffic
+    float traficCount(double time);
+};
+/*
+Thing 3) Cat
+5 properties:
+    1) number of paws
+    2) list of fur colours
+    3) gender
+    4) age
+    5) number of lives
+3 things it can do:
+    1) eat
+    2) sleep
+    3) do mew
+ */
+
+struct Cat
+{
+    //number of paws
+    int paw=4;
+    //list of fur colours
+    char colour;
+    //gender
+    bool gender = 0; //0 female, 1 male
+    //age
+    float age = 3.5f;
+    //number of lives
+    int liveNumber = 1;
+
+    //eat
+    void eat(char foodType);
+    //sleep
+    void sleep (float time);
+    //do mew
+    void mew (int count);
+};
+/*
+Thing 4) Spaceship
+5 properties:
+    1) orbit height
+    2) engine type
+    3) number of crew 
+    4) country of origin
+    5) name
+3 things it can do:
+    1) dock to space station
+    2) fly around planets
+    3) explore space
+ */
+
+struct spaceShip
+{
+    //orbit height
+    float orbitHeight = 1000;
+    //engine type
+    char engineType;
+    //number of crew 
+    int crewNum = 3;
+    //country of origin
+    char country;
+    //name
+    char name;
+    
+    //spaship's crew meber
+    struct crewMember
+    {
+        //member's personal id
+        int memberId = 0;
+        //member's name
+        std::string name;
+        //member's role
+        std::string jobRole = "scientist";
+        //member's weight
+        float weight = 70.0f;
+        //member's age
+        float age =33.3f;
+        
+        //do main job
+        void examineAnimal (int date, float time, Cat cat);
+        //record results
+        bool recordTest (int date, float time, int testNum=0);
+        //do additional mandatory job
+        void examineCrew (int date, float time, crewMember memberId);
+    };
+
+    //dock to space station
+    bool dock();
+    //fly around planets
+    void makeLoop(int planetNum = 3);
+    //explore space
+    bool takeOf(float startTime);
+
+};
+
+/*
+Thing 5) DAC
+5 properties:
+    1) samplerate
+    2) bit depth
+    3) SNR
+    4) amount of power consumed
+    5) dynamic range
+3 things it can do:
+    1) recieve PCM signal
+    2) correct errors 
+    3) produce analogue signal
+ */
+struct DAC
+{
+    //samplerate
+    int sampleRate = 48000;
+    //bit depth
+    int bitDepth = 24;
+    //SNR
+    int SNR = 112;
+    //amount of power consumed
+    float amountOfPoerwConsumed = 0.9f;
+    //dynamic range
+    double dynamicRange = 9.9;
+    
+    //recieve PCM signal
+    void readInput(int channelNum=0);
+    //correct errors 
+    bool checkError(long double sampleNum);
```

Comment:
don't use `long double`. it doesn't map to normal types on x86 chips.  use `double`
https://en.cppreference.com/w/cpp/language/types
scroll to the section on Floating Point Types
- [ ] revise